bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-metadata.js

```diff
@@ -0,0 +1,182 @@
+/*
+* Adds new blocks for metadata support
+* Note that the blocks are only support for GraalVM together with the JSScripting library
+*/
+
+import Blockly from 'blockly'
+import { javascriptGenerator } from 'blockly/javascript'
+import { addGetItemMetaConfigValue } from './utils'
+
+const unavailMsg = 'Metadata blocks aren\'t supported in "application/javascript;version=ECMAScript-5.1"'
+
+export default function (f7, isGraalJs) {
+  /*
+   Get the main value from the metadata namespace
+   item: either the item name string or the item object
+   namespace: string
+  */
+  Blockly.Blocks['oh_get_meta_value'] = {
+    init: function () {
+      this.appendValueInput('theItem')
+        .appendField('get metadata value of item')
+        .setCheck(['String', 'oh_item', 'oh_itemtype'])
+      this.appendValueInput('namespace')
+        .appendField('from namespace')
+      this.setInputsInline(true)
+      this.setOutput(true, 'String')
+      this.setColour(0)
+      this.setTooltip('get the main metadata value of item from that particular configuration namespace')
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/blockly/rules-blockly-metadata.html#get_meta_value')
+    }
+  }
+
+  javascriptGenerator['oh_get_meta_value'] = function (block) {
+    const theItem = javascriptGenerator.valueToCode(block, 'theItem', javascriptGenerator.ORDER_ATOMIC)
+    const namespace = javascriptGenerator.valueToCode(block, 'namespace', javascriptGenerator.ORDER_ATOMIC)
+
+    if (isGraalJs) {
+      return [`(items.metadata.getMetadata(${theItem}, ${namespace}) !== null) ? (items.metadata.getMetadata(${theItem}, ${namespace}).value) : 'undefined'`, javascriptGenerator.ORDER_CONDITIONAL]
```

Comment:
Can you please explain why you `null` check here and set `undefined` as a string?

Reply:
I had situations where the code would crash because the namespace didn't exist (or I misspelled it), so returning an undefined allows me to return the value undefined in case the namespace doesn't exist.